students/EricAdams/session04/trigrams.py

```diff
@@ -1,101 +1,166 @@
 import random
+# import string
+
+# raw_input_list = []
+input_string = []
+# initial_trigram_key = tuple()
+# output_list = []
+
+# Make these strings instead of lists for the str.translate method
+
+# List of chars to remove or replace with a space from input text
+# unwanted_punc = ['-',
+#                  "'",
+#                  ',',
+#                  '.',
+#                  ')',
+#                  '(',
+#                  '"',
+#                  '?',
+#                  ';',
+#                  '\n']
+
+# substitution_chars = [' ',
+#                       ' ',
+#                       ' ',
+#                       ' ',
+#                       ' ',
+#                       ' ',
+#                       ' ',
+#                       ' ',
+#                       ' ',
+#                       ' ']
+substitution_chars = ' ' * 10
+unwanted_punc = '-' + "'" + ',' + '.' + ')' + '(' + '"' + '?' + ';' + '\n'
+
+
+def replace_unwanted_punctuation(raw_input_string, unwanted_punc,
+                                 substitution_chars):
+    """ Build up an input string with unwanted punctuation replaced
+    in accordance with a string of chars
+    :param1 = a string with punctuation
+    :param2 = a string of unwanted punctuation
+    :param3 = a string of char to substitue for the unwanted punctuation
+    :return a string free of unwanted punctuation
+    """
+    # Now that the read in file is one string. Use str.translate
+    # for i, x in enumerate(raw_input_list):
+    #     if x in replace_punc:
+    #         input_list.append(' ')
+    #     else:
+    #         input_list.append(x)
+
+    # maketrans returns a table
+    transition_table = raw_input_string.maketrans(
+        unwanted_punc, substitution_chars)
+    input_string = raw_input_string.translate(transition_table)
+    return input_string
+
+
+def create_the_trigram(input_string):
+    """ Create a trigram from a list.
+    :param1 = a list
+    :return = a dictionary, (trigram)
+    """
+    trigram = {}
+    # we need a pair hence the len(input_list) - 2
+    #   instead of len(input_list) - 1
+    for i in range(len(input_string) - 2):
+        # must be a tuple in order to be a key in trigram
+        trigram_key = tuple(input_string[i:i + 2])
+        trigram_followers = input_string[i + 2]
+        # this is so nifty, I had to use it
+        trigram.setdefault(trigram_key, []).append(trigram_followers)
+    return trigram
+
+
+def initial_output_list(trigram):
+    """Initialize the output list with three elements derived from a
+    random trigram key
+    :param = dictionary, (trigram)
+    :return list
+    """
+    # Remove - 1
+    # random_number = random.randrange(0, len(trigram) - 1)
+    random_number = random.randrange(0, len(trigram))
+    for i, trigram_key in enumerate(trigram):
+        if i == random_number:
+            initial_trigram_key = trigram_key
+            break
+    output_list = list(initial_trigram_key) + trigram[initial_trigram_key]
```

Comment:
so here you are starting with your random pair of words -- perfect.

But then you're adding the list of following words -- which you don't want. You want only one of those words.

And now that you have an initial pair, you probably just want to use the regular loop to add the rest...